### ISSUE TITLE
ActiveStorage: foreign key constraints broke db:schema:load with mysql

### ISSUE THREAD

**Issue by andynu** [rails/rails#41491]
### Steps to reproduce

- rails new -d mysql astest
- cd astest;
- bin/rails active_storage:install
- bin/rails db:create
- bin/rails db:migrate
- bin/rails db:schema:load
```
rails aborted!
ActiveRecord::StatementInvalid: Mysql2::Error: Cannot drop table 'active_storage_blobs' referenced by a foreign key constraint 'fk_rails_993965df05' on table 'active_storage_variant_records'.
/home/andy/tmp/ras-WU89/astest/db/schema.rb:25:in `block in <main>'
/home/andy/tmp/ras-WU89/astest/db/schema.rb:13:in `<main>'
/home/andy/tmp/ras-WU89/astest/bin/rails:5:in `<top (required)>'
/home/andy/tmp/ras-WU89/astest/bin/spring:10:in `block in <top (required)>'
/home/andy/tmp/ras-WU89/astest/bin/spring:7:in `tap'
/home/andy/tmp/ras-WU89/astest/bin/spring:7:in `<top (required)>'

Caused by:
Mysql2::Error: Cannot drop table 'active_storage_blobs' referenced by a foreign key constraint 'fk_rails_993965df05' on table 'active_storage_variant_records'.
/home/andy/tmp/ras-WU89/astest/db/schema.rb:25:in `block in <main>'
/home/andy/tmp/ras-WU89/astest/db/schema.rb:13:in `<main>'
/home/andy/tmp/ras-WU89/astest/bin/rails:5:in `<top (required)>'
/home/andy/tmp/ras-WU89/astest/bin/spring:10:in `block in <top (required)>'
/home/andy/tmp/ras-WU89/astest/bin/spring:7:in `tap'
/home/andy/tmp/ras-WU89/astest/bin/spring:7:in `<top (required)>'
Tasks: TOP => db:schema:load
(See full trace by running task with --trace)
```

### Expected behavior
It should reload the schema without error.

### Actual behavior
It did not reload the schema, instead it had an error on deleting an active_storage_blobs foreign key constraint.

### System configuration
**Rails version**: 6.1.3

**Ruby version**: 2.7.2p137

**Comment by intrip** [comment]
This looks like just a FK problem and not related to ActiveStorage. You should be able to fix the error by calling `rails db:drop` before `rails db:schema:load`

**Comment by andynu** [comment]
It was a FK added for active storage.

I agree that calling `rails db:drop` does remove the FK and allow `rails
db:schema:load` to work. That is a fine workaround.

However, the introduction of a FK that is not handled by `db:schema:load`
itself is new and undesirable. That perhaps `db:schema:load` should handle
the ordering of the tables being dropped with regard to the foreign keys
that were introduced, or should be automatically preceded with a `db:drop`.
In all prior versions to this FK being introduced `db:schema:load` was
sufficient.

My personal experience here is that for years I've been running
db:schema:load at the beginning of my CI suite to ensure the database is
what it should be. And recently CI across all of my rails projects broke
because of this. I have since disabled the foreign_key_checks in mysql on
the test databases, which is another workaround.

**Comment by andynu** [comment]
It is still reproducable in 6-1-stable.

Gemfile:
```
# frozen_string_literal: true

source "https://rubygems.org"

git_source(:github) { |repo| "https://github.com/#{repo}.git" }

gem 'rails', github: 'rails/rails', branch: '6-1-stable'
```

Reproduction:
```
+ bundle
+ bundle exec rails new -d mysql --skip-javascript astest 2>&1 2>/dev/null
+ cd astest
+ bundle exec rails active_storage:install
Copied migration 20210627195317_create_active_storage_tables.active_storage.rb from active_storage
+ bundle exec rails db:prepare
== 20210627195317 CreateActiveStorageTables: migrating ========================
-- create_table(:active_storage_blobs, {})
   -> 0.0561s
-- create_table(:active_storage_attachments, {})
   -> 0.0658s
-- create_table(:active_storage_variant_records, {})
   -> 0.0607s
== 20210627195317 CreateActiveStorageTables: migrated (0.1831s) ===============

+ bundle exec rails db:schema:load
rails aborted!
ActiveRecord::StatementInvalid: Mysql2::Error: Cannot drop table 'active_storage_blobs' referenced by a foreign key constraint 'fk_rails_993965df05' on table 'active_storage_variant_records'.
/home/andy/tmp/rails-issue-41491-BGIs/astest/db/schema.rb:25:in `block in <main>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/db/schema.rb:13:in `<main>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/rails:5:in `<top (required)>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:10:in `require'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:10:in `block in <top (required)>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:7:in `tap'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:7:in `<top (required)>'

Caused by:
Mysql2::Error: Cannot drop table 'active_storage_blobs' referenced by a foreign key constraint 'fk_rails_993965df05' on table 'active_storage_variant_records'.
/home/andy/tmp/rails-issue-41491-BGIs/astest/db/schema.rb:25:in `block in <main>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/db/schema.rb:13:in `<main>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/rails:5:in `<top (required)>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:10:in `require'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:10:in `block in <top (required)>'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:7:in `tap'
/home/andy/tmp/rails-issue-41491-BGIs/astest/bin/spring:7:in `<top (required)>'
Tasks: TOP => db:schema:load
(See full trace by running task with --trace)
```

**Comment by intrip** [comment]
@andynu I think the purpose of `schema:load` is to run it from an empty new database hence in order to make sure it works you need to re-create the database at first.

**Comment by andynu** [comment]
Hi Jacopo, I can't speak to original intentions, but I can speak to prior
behavior. Up until this foreign key was added it worked to reset the schema
regardless of if tables were there or not. I know I found that behavior
useful.

I'm in favor of the former behavior. If it could be restored by re-ordering
the statements, or calling db:drop automatically prior to db:schema:load, I
think that would be great.
